### ISSUE TITLE
Not converging?

### ISSUE THREAD

**Issue by kimren227** [BunnySoCrazy/SECAD-Net#1]
Hi,

Thanks for this exciting work and for releasing the code. I tried to train the network following the instructions in the README, however, it seems that the network is not converging (see Figs attached). Any suggestions on how to train the network correctly?

Thanks,

Daxuan

<img width="304" alt="image" src="https://github.com/BunnySoCrazy/SECAD-Net/assets/8415809/ec84491e-49ad-4190-98b3-2e6534c5d96b">
<img width="1059" alt="image" src="https://github.com/BunnySoCrazy/SECAD-Net/assets/8415809/e84ef946-9854-4410-8d56-0c7717dc49d9">
<img width="1061" alt="image" src="https://github.com/BunnySoCrazy/SECAD-Net/assets/8415809/38ff5899-c46d-48f6-822b-0aabfa9c775d">

<img width="311" alt="image" src="https://github.com/BunnySoCrazy/SECAD-Net/assets/8415809/19db7fc3-ebd9-41f0-9a6e-c62bcd05637d">
<img width="291" alt="image" src="https://github.com/BunnySoCrazy/SECAD-Net/assets/8415809/9a023357-5121-4cdf-91b3-6fd09c2656ea">

**Comment by BunnySoCrazy** [comment]
I'll recheck the training code. Before that, please use the pre-training parameters in this [link](https://drive.google.com/file/d/1uP-AoCqAMOQ9Q2W4fwk0iN5sGCY6UC4x/view?usp=share_link) to fine-tune the shape. Hope this helps.

**Comment by BunnySoCrazy** [comment]
I discovered that the original training code was loading the test dataset instead of the training dataset. I have rectified the issue. I sincerely appreciate your valuable feedback on my mistake.

**Comment by kimren227** [comment]
Thanks for the swift reply, the provided pre-trained model works well for fine-tuning. However, I still facing the same issue during training after pulling the latest change. Looking forward to the code fix. If there is anything I can help with, do let me know.
<img width="355" alt="image" src="https://github.com/BunnySoCrazy/SECAD-Net/assets/8415809/a0275100-9b78-42f1-844c-1481901bf0e6">

**Comment by hmrishavbandy** [comment]
I think the loss is reversed 🤔 

https://github.com/BunnySoCrazy/SECAD-Net/blob/af468179b2b0300d42a6a030290fd0c50a592151/trainer/loss.py#L18C1-L18C60

**Comment by BunnySoCrazy** [comment]
I've rerun the training code and everything seems to be working fine, so I'm going to close this issue now.
